core/app/components/DataTable/PubsDataTable/PubsDataTableClient.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+
+import React, { useMemo, useState } from "react";
+import { useQueryStates } from "nuqs";
+
+import type { ProcessedPub } from "contracts";
+import type { DataTableRowAction } from "ui/data-table-paged";
+import { DataTable, useDataTable } from "ui/data-table-paged";
+
+import type { PubForTable } from "./types";
+import { client } from "~/lib/api";
+import { useCommunity } from "../../providers/CommunityProvider";
+import { getColumns } from "./columns";
+import { dataTableParsers, getFilterParamsFromSearch } from "./validations";
+
+interface WithPromises {
+	promises: Promise<[number, PubForTable[]]>;
+}
+interface WithData {
+	total: number;
+	data: ProcessedPub<{ withPubType: true }>[];
+}
+type PubsDataTableProps = {
+	perPage: number;
+} & (WithPromises | WithData);
+
+export const PubsDataTable = ({ perPage, ...props }: PubsDataTableProps) => {
```

Comment:
made this table able to accept promises or data. it was difficult to get react query to return a promise to keep the interface exactly the same

Seems to work okay with the query params, although I'm not sure we want to use query params here, maybe just state instead? query params would make it difficult to render two of the same component on the page (not that we are doing that for pub tables anywhere atm)

![image](https://github.com/user-attachments/assets/cfea9640-c037-4447-85fd-a202641bcf62)

Reply:
yes i think we should use state for the react-query variation. although i think it could be fine if we used the `shallow: true` in that case (this prevents the page from reloading all the data on searchParam change). if we do though, we should prefix the parameters with something st we can distinguish different usages of the component on the same page.

but my gut feel is to just leave it out, just use state for the react query version!

Reply:
the main consideration i would have: in which situations would we use the client component version, _and_ would want the state of the page to be preserved on eg reload. currently i don't see that many cases for it